app/src/androidTest/java/com/duckduckgo/app/browser/downloader/UriUtilsFilenameExtractorTest.kt

```diff
@@ -69,6 +69,47 @@ class UriUtilsFilenameExtractorTest {
         assertEquals("realFilename.jpg", extractionResult.filename)
     }
 
+    @Test
+    fun whenUrlContainsFilenameButContainsPathSegmentsWhichLookLikeAFilenameThenFilenameShouldBeExtracted() {
+        val url = "https://foo.example.com/path/dotted.path/b/b1/realFilename.jpg"
+        val mimeType: String? = null
+        val contentDisposition: String? = null
+
+        val extractionResult = testee.extract(buildPendingDownload(url, contentDisposition, mimeType))
+        assertTrue(extractionResult is FilenameExtractor.FilenameExtractionResult.Extracted)
+
+        extractionResult as FilenameExtractor.FilenameExtractionResult.Extracted
+        assertEquals("realFilename.jpg", extractionResult.filename)
+    }
+
+    @Test
+    @SdkSuppress(minSdkVersion = 22)
+    fun whenUrlContainsAmbiguousFilenameButContainsPathSegmentsWhichLookLikeAFilenameAndMimeTypeProvidedThenFilenameShouldBeExtracted() {
+        val url = "https://foo.example.com/path/dotted.path/b/b1/realFilename"
+        val mimeType: String? = "image/jpeg"
+        val contentDisposition: String? = null
+
+        val extractionResult = testee.extract(buildPendingDownload(url, contentDisposition, mimeType))
+        assertTrue(extractionResult is FilenameExtractor.FilenameExtractionResult.Extracted)
+
+        extractionResult as FilenameExtractor.FilenameExtractionResult.Extracted
+        assertEquals("realFilename.jpg", extractionResult.filename)
+    }
+
+    @Test
+    @SdkSuppress(minSdkVersion = 22)
```

Comment:
ditto

Reply:
Since this test contains a guess with an extension which satisfies the MIME type, I believe that any SDK version will recognize this (versus the other test where the extension was ambiguous). I'll remove this and see if it still passes!

Reply:
Makes sense 👍

Reply:
Removed this and it passed 🎉